app/src/main/java/com/example/fisrtapplication/fragments/ProfileFragment.java

```diff
@@ -0,0 +1,254 @@
+        nameRef.addListenerForSingleValueEvent(new ValueEventListener() {
+            @Override
+            public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
+                String value = dataSnapshot.getValue(String.class);
+                profileName.setText(value);
+            }
+
+            @Override
+            public void onCancelled(@NonNull DatabaseError databaseError) {
+                Toast.makeText(getContext(), "Failed to load name", Toast.LENGTH_SHORT).show();
+            }
+        });
+    }
+
+    private void updateEmail() {
+        String email = newEmail.getText().toString();
+//        if (validateEmail(email)) {
+            fuser.updateEmail(email)
+                    .addOnCompleteListener(task1 -> {
+                        if (task1.isSuccessful()) {
+                            Toast.makeText(getContext(), "User email address updated.",
+                                    Toast.LENGTH_SHORT).show();
+                            newEmail.getText().clear();
+                            showEmail();
+                        }
+                    });
+//        }
+    }
+
+    private void updateName() {
+        String name = newName.getText().toString();
+        if (validateName(name)) {
+            reference.child("name").setValue(name);
+            Toast.makeText(getContext(), "Name updated", Toast.LENGTH_SHORT).show();
+            newName.getText().clear();
+            showName();
+        }
+    }
+
+    private void setupViews(View inflate) {
+        storageReference = FirebaseStorage.getInstance().getReference("uploads");
+        fuser = FirebaseAuth.getInstance().getCurrentUser();
+        reference = FirebaseDatabase.getInstance().getReference("users").child(fuser.getUid());
+        uploadImgButton = inflate.findViewById(R.id.new_image_button);
+        newEmailButton = inflate.findViewById(R.id.new_email_button);
+        newNameButton = inflate.findViewById(R.id.new_name_button);
+        newName = inflate.findViewById(R.id.name_new);
+        newEmail = inflate.findViewById(R.id.email_new);
+        profileName = inflate.findViewById(R.id.username);
+        profileEmail = inflate.findViewById(R.id.user_email);
+        profileImage = inflate.findViewById(R.id.image_profile);
+        urlRef = reference.child("imageURL");
+        nameRef = reference.child("name");
+        emailRef = reference.child("email");
+    }
+
+    private void openImage() {
+        Intent intent = new Intent();
+        intent.setType("image/*");
+        intent.setAction(Intent.ACTION_GET_CONTENT);
+        startActivityForResult(intent, IMAGE_REQUEST);
+    }
+
+    private String getFileExtension(Uri uri) {
+        ContentResolver contentResolver = Objects.requireNonNull(getContext()).getContentResolver();
+        MimeTypeMap mimeTypeMap = MimeTypeMap.getSingleton();
+        return mimeTypeMap.getExtensionFromMimeType(contentResolver.getType(uri));
+    }
+
+    private void uploadImage() {
```

Comment:
This needs to be refactored, complexity is too high and it becomes incomprehensible.